src/org/zaproxy/zap/extension/jython/JythonOptionPanel.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2013 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.jython;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.NotDirectoryException;
+
+import javax.swing.GroupLayout;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JLabel;
+import javax.swing.JTextField;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.view.AbstractParamPanel;
+
+import org.python.google.common.base.Strings;
+
+public class JythonOptionPanel extends AbstractParamPanel implements ActionListener {
+
+	private static final long serialVersionUID = -2690686914494943483L;
+
+	private JTextField modulesPathTextField;
+	private JButton pathChooseButton;
+
+	public JythonOptionPanel() {
+		super();
+		this.initComponents();
+	}
+
+	@Override
+	public String getHelpIndex() {
+		return null;
+	}
+
+	@Override
+	public void initParam(Object object) {
+		JythonOptionParam jythonOptionParam = ((OptionsParam) object).getParamSet(JythonOptionParam.class);
+		this.modulesPathTextField.setText(Strings.nullToEmpty(jythonOptionParam.getModulePath()));
+	}
+
+	@Override
+	public void saveParam(Object object) throws Exception {
+		JythonOptionParam jythonOptionParam = ((OptionsParam) object).getParamSet(JythonOptionParam.class);
+		jythonOptionParam.setModulePath(Strings.nullToEmpty(this.modulesPathTextField.getText()));
+	}
+
+	@Override
+	public void validateParam(Object object) throws Exception {
+		String modulesPathString = this.modulesPathTextField.getText();
+		if (false == Strings.isNullOrEmpty(modulesPathString)) {
+			File modulesPath = new File(modulesPathString);
+			if (false == modulesPath.exists()) {
+				throw new NoSuchFileException(
+						Constant.messages.getString("jython.options.error.modulepath.notexist", modulesPath));
+			}
+			if (false == modulesPath.isDirectory()) {
+				throw new NotDirectoryException(
+						Constant.messages.getString("jython.options.error.modulepath.notdirectory", modulesPath));
+			}
+		}
+	}
+
+	private void initComponents() {
```

Comment:
Could this method be moved to after the constructor (or merged into)? Easier to read the code, no need to search for it.